delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/ResourcesDummy.scala

```diff
@@ -23,88 +24,112 @@ import monix.bio.{IO, Task, UIO}
-      eval(CreateResource(id, project.ref, schema, source, compacted, expanded, caller))
-    }
+    for {
+      project                  <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      jsonld                   <- sourceAsJsonLD(project, source)
+      (id, compacted, expanded) = jsonld
+      res                      <- eval(CreateResource(id, projectRef, schema, source, compacted, expanded, caller), project.apiMappings)
+    } yield res
 
   override def create(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schema: ResourceRef,
       source: Json
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
-      eval(CreateResource(id, project, schema, source, compacted, expanded, caller))
-    }
+    for {
+      jsonld               <- sourceAsJsonLD(id, source)
+      (compacted, expanded) = jsonld
+      project              <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      res                  <- eval(CreateResource(id, projectRef, schema, source, compacted, expanded, caller), project.apiMappings)
+    } yield res
 
   override def update(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       rev: Long,
       source: Json
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
-      eval(UpdateResource(id, project, schemaOpt, source, compacted, expanded, rev, caller))
-    }
+    for {
+      jsonld               <- sourceAsJsonLD(id, source)
+      (compacted, expanded) = jsonld
+      project              <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      mapping               = project.apiMappings
+      res                  <- eval(UpdateResource(id, projectRef, schemaOpt, source, compacted, expanded, rev, caller), mapping)
+    } yield res
 
   override def tag(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       tag: Label,
       tagRev: Long,
       rev: Long
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    eval(TagResource(id, project, schemaOpt, tagRev, tag, rev, caller))
+    for {
+      project <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      res     <- eval(TagResource(id, projectRef, schemaOpt, tagRev, tag, rev, caller), project.apiMappings)
+    } yield res
 
   override def deprecate(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       rev: Long
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    eval(DeprecateResource(id, project, schemaOpt, rev, caller))
+    for {
+      project <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
```

Comment:
When deprecating, the project could be deprecated, it is not a problem no ?

Also like schemas and other stuff, we may want to fetch a specific revision of the project to avoid mutability if the api mappings changes

Reply:
No, deprecating a resource is not allowed when project is deprecated (at least not in v.1.4.x)

No, there is no immutability in terms of apimappings. This is metadata what we are talking about